### PR TITLE
Add sphinx documentation

### PR DESCRIPTION
**Pull request by antirotor**
This is an effort to add at least auto-generated API documentation using Sphinx. This PR is:

- adding necessary dependencies
- providing basic infrastructure for Sphinx
- providing some initial `rst` files to generate documentations
- dealing with some docstring style issues

**Testing**

1) First install all dependencies: `poetry install`
2) Run *apidoc*:
```sh
cd ./docs/
poetry run sphinx-apidoc -f -o ../docs/source/ ../ ../tests ../*setup*
```
3) Generate html:
```sh
poetry run sphinx-build -M html .\source .\build
```
or use provided batch scripts:

**check_docstrings.bat** - will run `pydocstyle` against the sources to print out any issues
**make_api.bat** - will run apidocs
**make_html.bat** - will build html documentation.

Result can be found in `./docs/build/html` - just open `index.html` in a browser.

> [!NOTE]
> Some `.rst` files should be tweaked, some docstring syntax should be checked (build is producing a lot of errors)

### CODE REVIEW

File: ayon_api/server_api.py [hunk 1]
```diff
@@ -2082,7 +2082,8 @@ def get_addons_info(self, details=True):
     def get_addon_url(self, addon_name, addon_version, *subpaths):
         """Calculate url to addon route.
 
-        Example:
+        Examples:
+
```
Comment (iLLiCiTiT) [inline ayon_api/server_api.py:2086]: Why are there empty lines after each double colon?

File: ayon_api/server_api.py [hunk 2]
```diff
@@ -2322,7 +2323,8 @@ def get_dependency_packages(self):
         To download dependency package, use 'download_dependency_package'
         method and pass in 'filename'.
 
-        Example data structure:
+        Example data structure::
```
Comment (iLLiCiTiT) [inline ayon_api/server_api.py:2326]: Why double double colon?

File: ayon_api/server_api.py [hunk 3]
```diff
@@ -367,10 +367,10 @@ class ServerAPI(object):
         sender (Optional[str]): Sender of requests. Used in server logs and
             propagated into events.
         ssl_verify (Union[bool, str, None]): Verify SSL certificate
-            Looks for env variable value 'AYON_CA_FILE' by default. If not
+            Looks for env variable value ``AYON_CA_FILE`` by default. If not
```
Comment (iLLiCiTiT) [inline ayon_api/server_api.py:370]: Why double ` `` `?

File: ayon_api/server_api.py (hunk 2 above)
Comment (antirotor) [reply ayon_api/server_api.py:2326]: That is actually RST syntax to format everything as a code block following those double double colons:

In markdown, equivallent is:

---

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

**Example data structure:**
    {
        "foo": "bar"
    }

---

With double double colons you'll get:

---
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

**Example data structure:**
```
    {
        "foo": "bar"
    }
```

Comment (antirotor) [reply ayon_api/server_api.py:2326]: For more, see  https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks

File: ayon_api/server_api.py (hunk 3 above)
Comment (antirotor) [reply ayon_api/server_api.py:370]: that relates how it would be formated, similar to markdown. It will do `foo` instead of 'foo'

**Review by mkolar — APPROVED** [review]

**Pull request merged**